subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/ExecuteActionsTaskExecuter.java

```diff
@@ -84,12 +84,11 @@ public void execute(TaskInternal task, TaskStateInternal state, TaskExecutionCon
     private GradleException executeActions(TaskInternal task, TaskStateInternal state, TaskExecutionContext context) {
         LOGGER.debug("Executing actions for {}.", task);
         final List<ContextAwareTaskAction> actions = new ArrayList<ContextAwareTaskAction>(task.getTaskActions());
-        int actionNumber = 1;
         for (ContextAwareTaskAction action : actions) {
             state.setDidWork(true);
             task.getStandardOutputCapture().start();
             try {
-                executeAction("Execute task action " + actionNumber + "/" + actions.size() + " for " + task.getPath(), task, action, context);
+                executeAction(action.getDisplayName(), task, action, context);
```

Comment:
One nitpicky thing, since this includes "Execute ..." in its name, it's awkward to display this name anywhere else that's not "executing" the action.

e.g., it might be useful in the exceptions/logs below to include the action's display name to narrow problems down to particular actions.